lib/exabgp/bgp/message/update/attribute/bgpls/link/sradjlan.py

```diff
@@ -66,29 +65,28 @@ def unpack (cls,data,length):
 		# *  A 4 octet index defining the offset in the SID/Label space
 		# 	 advertised by this router using the encodings defined in
 		#  	 Section 3.1.  In this case V and L flags MUST be unset.
-		sids = []
 		raw = []
 		while data:
 			# Range Size: 3 octet value indicating the number of labels in
 			# the range.
-			if int(flags.flags['V']) and int(flags.flags['L']):
+			if int(flags['V']) and int(flags['L']):
 				b = BitArray(bytes=data[:3])
 				sid = b.unpack('uintbe:24')[0]
 				data = data[3:]
-				sids.append(sid)
-			elif (not flags.flags['V']) and \
-				(not flags.flags['L']):
+			elif (not flags['V']) and (not flags['L']):
 				sid = unpack('!I',data[:4])[0]
 				data = data[4:]
-				sids.append(sid)
 			else:
 				raw.append(hexstring(data))
 				break
-
-		return cls(flags=flags, sids=sids, weight=weight,undecoded=raw)
+		cls._sr_adj_lan_sids.append(
+			{'flags': flags, 'weight': weight, 'system-id': system_id, 'sid': sid, 'undecoded': raw}
+		)
+		return cls()
 
 	def json (self,compact=None):
-		return ', '.join(['"sr-adj-lan-flags": {}'.format(self.flags.json()),
-			'"sids": {}'.format(json.dumps(self.sids)),
-			'"undecoded-sids": {}'.format(json.dumps(self.undecoded)),
-			'"sr-adj-lan-weight": {}'.format(json.dumps(self.weight))])
+		return '"sr-adj-lan-sids": {}'.format(json.dumps(self.sr_adj_lan_sids))
```

Comment:
@thomas-mangin do you think it makes sense for us to move to use` json.dumps()` as @tomjshine is doing here? At the time I didn't do it probably to be consistent with some other portions of the code :|

Reply:
I have no objections